src/Config.h

```diff
@@ -0,0 +1,1488 @@
+
+/**
+ * @file Config.h
+ *
+ * @brief Centralized configuration management with JSON storage
+ */
+
+#pragma once
+
+#include "Logger.h"
+#include "defaults.h"
+#include "hardware/Relay.h"
+#include "hardware/Switch.h"
+#include <ArduinoJson.h>
+#include <LittleFS.h>
+
+class Config {
+    public:
+        /**
+         * @brief Initialize the configuration system
+         *
+         * @return true if successful, false otherwise
+         */
+        bool begin() {
```

Comment:
I think I would move the initialization to the constructor. In the `main.cpp` you would then just keep a pointer around (`std::shared_ptr<Config> config`) and create the object when due `config = std::make_shared<Config>()` which initializes itself and loads defaults.

The benefit is that you cannot interact with the object without being initialized - something that can happen right now.